### PR TITLE
Fix Gradle build on kotlin 1.7.0

### PR DESCRIPTION
**Pull request by nulls**

### CODE REVIEW

**Review by petertrr — APPROVED** [review]

**Review by petertrr — CHANGES_REQUESTED** [review]
Oh, sorry, could you please also update cache keys for konan in all github workflow files to contain `-1.7.0`?

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 plugins {
-    `kotlin-dsl`
+    // workaround https://github.com/gradle/gradle/issues/16345
```
Comment (akuleshov7) [inline buildSrc/build.gradle.kts:2]: btw - the issue is from the February... 
Very strange

**Review by akuleshov7 — APPROVED** [review]

File: buildSrc/build.gradle.kts (hunk 1 above)
Comment (nulls) [reply buildSrc/build.gradle.kts:2]: The issue is from February 2021. it's long live issue.
I think that a fix is possible only in Gradle 8.
But comment about `Kotlin 1.7.0` is from June 9th (release date :))

**Comment by nulls** [comment]
> also maybe upgrade:
> 
> ```toml
> # kx.datetime 0.3.3 is built with Kotlin 1.7, so currently not compatible with our build
> kotlinx-datetime = { require = "0.3.2", rejects = "0.3.3" }
> ```

done

**Review by petertrr — APPROVED** [review]

**Pull request merged**